keras/layers/core.py

```diff
@@ -685,13 +685,43 @@ def __init__(self, dims, **kwargs):
         super(Reshape, self).__init__(**kwargs)
         self.dims = tuple(dims)
 
+    def _fix_unknown_dimension(self, input_shape, output_shape):
+        """
+        A near direct port of the internal numpy function _fix_unknown_dimension
+        in numpy/core/src/multiarray/shape.c
+        """
```

Comment:
A docstring that explicitly describes what this function does would be helpful.

Also, style nitpick: rather than `"` and `"""`, `'` and `'''` is used throughout the codebase.